### PR TITLE
Lagrange Vector Transfer Support

### PR DESCRIPTION
**Pull request by EdwardPalmer99**
This PR makes it possible to transfer lagrange vectors between a MOOSE app and a miniapp. 

Changes:
1. Added a new auxkernel 'VectorVariableFromComponentsAux'. This allows the user to create a Lagrange vector from three standard Lagrange variables of the same order. There was no existing MOOSE auxkernel to do this.
2. Added a new postprocessor 'ElementVectorL2Difference' to compute the L2-norm difference between two Lagrange vectors. This is useful for quickly comparing lagrange vectors without separating them into individual components and comparing those.
3. Added a new action 'AddTransferLagrangeVecAction'. The user can use this to create an auxiliary Lagrange vector of the desired order. It will stealthily create three hidden standard Lagrange variables of the same order using the name of the Lagrange vector appended with "_x", "_y", "_z". This allows the user to not have to write-out explicitly the corresponding standard Lagrange variables required for transfers

Tests:
1. Added a test for the VectorVariableFromComponentsAux auxkerenel
2. Added a test for the ElementVectorL2Difference postprocessor
3. Added a one-way test for the Lagrange vector transfer

Notes on the transfer:
1. Use the AddTransferLagrangeVecAction to setup the auxiliary variable you wish to transfer
2. Do something with the Lagrange vector you've created
3. When ready to transfer, you need to update the _x, _y, _z component stealth Lagrange variables. To do this, use the VectorVariableComponentAux kernel for each component
4. In the sub-app (or main app depending on which way you're transferring), create a corresponding Lagrange vector to transfer to using the new action
5. Set each _x, _y, _z component in the MultiApp transfer
6. Now update the corresponding Lagrange vector from its stealth components using the new VectorVariableFromComponentsAux auxiliary kernel

### CODE REVIEW

File: src/problem/MFEMProblem.C [hunk 1]
```diff
@@ -259,17 +263,31 @@ MFEMProblem::addKernel(const std::string & kernel_name,
...
 MFEMProblem::addAuxKernel(const std::string & kernel_name,
                           const std::string & name,
                           InputParameters & parameters)
 {
-  FEProblemBase::addUserObject(kernel_name, name, parameters);
-  MFEMAuxSolver * mfem_auxsolver(&getUserObject<MFEMAuxSolver>(name));
-  mfem_problem_builder->AddPostprocessor(name, mfem_auxsolver->getAuxSolver(), true);
-  mfem_auxsolver->storeCoefficients(_coefficients);
+  // NB: - slightly crude test.
+  bool is_mfem_auxkernel = (strncmp(kernel_name.c_str(), "MFEM", 4) == 0);
```
Comment (alexanderianblair) [inline src/problem/MFEMProblem.C:278]: This is indeed fragile - should have a check on the base class, at least.

Comment (EdwardPalmer99) [reply src/problem/MFEMProblem.C]: This has been resolved in the latest commit by registering MFEMAuxKernel as a base class. Now in the MFEMProblem addAuxKernel method, we can determine whether the auxkernel is an MFEM auxsolver or a MOOSE auxkernel.

File: include/actions/AddTransferLagrangeVecAction.h
[the remark this answers is not in the document]
Comment (EdwardPalmer99) [reply include/actions/AddTransferLagrangeVecAction.h:1]: Stripped-out TransferLagrangeVec action.

**Review by alexanderianblair — APPROVED** [review]
Great - all comments resolved! Thanks

**Pull request merged**